analyzers/src/SonarAnalyzer.CSharp/Rules/CollectionsShouldImplementGenericInterface.cs

```diff
@@ -34,34 +36,39 @@ public sealed class CollectionsShouldImplementGenericInterface : SonarDiagnostic
         internal const string DiagnosticId = "S3909";
         private const string MessageFormat = "Refactor this collection to implement '{0}'.";
 
-        private static readonly DiagnosticDescriptor rule =
+        private static readonly DiagnosticDescriptor Rule =
             DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
 
-        private static readonly Dictionary<string, KnownType> nongenericToGenericMapping =
-            new Dictionary<string, KnownType>
+        private static readonly Dictionary<string, KnownType> NongenericToGenericMapping =
+            new()
```

Comment:
I'd be also fine with having the `new()` on the previous line. Just to save space (it's boring scaffolding)